In [10]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression
from sklearn.linear_model import SGDRegressor

In [5]:
data = pd.read_csv("data/raw.csv")

In [6]:
data.head()

,Unnamed: 0,Address,Zip,Price,Area,Room,Lon,Lat
0,1,"Blasiusstraat 8 2, Amsterdam",1091 CR,685000.0,64,3,4.907736,52.356157
1,2,"Kromme Leimuidenstraat 13 H, Amsterdam",1059 EL,475000.0,60,3,4.850476,52.348586
2,3,"Zaaiersweg 11 A, Amsterdam",1097 SM,850000.0,109,4,4.944774,52.343782
3,4,"Tenerifestraat 40, Amsterdam",1060 TH,580000.0,128,6,4.789928,52.343712
4,5,"Winterjanpad 21, Amsterdam",1036 KN,720000.0,138,5,4.902503,52.410538


In [7]:
data.shape

(924, 8)

From the above assessment, the dataset has 924 rows and 8 features (columns). The features are:
- Address
- Zip code
- Price
- Area
- Number of rooms
- Longitude
- Latitude

In [19]:
# Splitting train-test set iteratively
data = data.dropna()
split = np.arange(0.8, 0.2, -0.05)
X = data.iloc[:,4:]
y = data.iloc[:,3]
X

,Area,Room,Lon,Lat
0,64,3,4.907736,52.356157
1,60,3,4.850476,52.348586
2,109,4,4.944774,52.343782
3,128,6,4.789928,52.343712
4,138,5,4.902503,52.410538
...,...,...,...,...
919,117,1,4.927757,52.354173
920,72,3,4.890612,52.414587
921,51,3,4.856935,52.363256
922,113,4,4.965731,52.375268


In [22]:
def training(split):
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=split, random_state=42)
    lin_reg = LinearRegression().fit(X_train, y_train)
    #log_reg = LogisticRegression().fit(X_train, y_train)
    #sgd_reg = SGDRegressor(max_iter=1000)
    #sgd_reg.fit(X_train, y_train)
    

LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
LinearRegression()
